api/src/main/java/org/openmrs/ObsReferenceRange.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs;
+
+import org.hibernate.search.annotations.DocumentId;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.Table;
+
+/**
+ * ObsReferenceRange is typically a reference range of an Observation 
+ * The reference range is created at the point of recording {@link Obs}
+ *
+ * @since 2.7.0
+ */
+@Entity
+@Table(name = "obs_reference_range")
+public class ObsReferenceRange extends BaseReferenceRange implements java.io.Serializable {
+	
+	private static final long serialVersionUID = 473299L;
+
+	// Fields
+	@DocumentId
+	@Id
+	@Column(name = "obs_reference_range_id")
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	private Integer obsReferenceRangeId;
+
+	@JoinColumn(name = "obs_id", nullable = false)
+	private Obs obs;
+
+	/** Default constructor */
```

Comment:
What value does the above comment add? Did you get a chance to look at this? https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/

Reply:
The main reason I am adding comments in most places is because it seemed to be a practice at openMRS. For instance, the article you just mentioned suggests that we shouldn't add comments on every class, but we do. 

```
I don’t like blocks of header comments at the top of every file, method or class.
```

Reply:
Comments on classes are fine only when they add value. But where they do not, i would not recommend adding them because they increase our code maintenance burden. For instance, a comment telling one that this is default constructor would be of value to only someone who is using the openmrs code base to learn the basics of Java, as used in demonstration blocks of code in classes. 😊